### ISSUE TITLE
As a super user, I can customize the branding slightly (VCMA)

### ISSUE THREAD

**Issue by ggordn3r** [iambateman/RankIt#102]
Request from VoterChoice Massachusetts:

_Branding it with our colors and logo. It needs to be specific to MA. Logo block, custom color pallet, powered by FV is okay, customized the contact at the bottom._

What we propose to do out of this:
- allow super-users to change _some_ colors (e.g. the bar)
- allow co-branding alongside RankIt logo

**Comment by proggeramlug** [comment]
This is implemented and can be tested now.

Example: https://rankit.skelpo.com/vote/ICkHuqpt8zcol1mGGxQb

**Comment by ggordn3r** [comment]
I'm unable to test this fully because I am unable to vote--see #109

**Comment by ggordn3r** [comment]
I was able to test this and it works well. I am concerned, though, that it could result in a new request for a "preview mode" of the custom HTML.

If that does happen, the best way to handle it is: add a "Preview" option to the page where the "Publish" option resides while the poll is in Draft.

**Comment by ggordn3r** [comment]
Sorry--ignore the shuffle. I accidentally moved the wrong one.

**Comment by proggeramlug** [comment]
okay, still to be tested :)

**Comment by ggordn3r** [comment]
Tested this, but I can't see the results to confirm that the bar has changed color in the link you provided. See screenshot.

![Screenshot (265)](https://user-images.githubusercontent.com/10144772/80524197-3e652000-895d-11ea-8af6-52f788acbc14.png)

**Comment by proggeramlug** [comment]
Reload, you should see this now: 
![Screen Shot 2020-04-28 at 2 45 26 PM](https://user-images.githubusercontent.com/8865968/80525250-ec24fe80-895e-11ea-8e10-6ad36ca04085.png)

**Comment by ggordn3r** [comment]
Now the success page is right, but I still can't see results. I think you have it set to hold them in your settings with #106! 😆

**Comment by proggeramlug** [comment]
now! :)

**Comment by ggordn3r** [comment]
Hmm... I see the logo, but the colors don't look any different:

![Screenshot (274)](https://user-images.githubusercontent.com/10144772/80754201-9f772a00-8afc-11ea-9549-41f4a1236398.png)

**Comment by proggeramlug** [comment]
Alright, check now. I had implemented it for the "vote" page only.

We are for now only changing the color of the header. Should we apply those colors to the buttons as well?

**Comment by ggordn3r** [comment]
Ah, looking back at the description, I see that I didn't define this very well. Sorry about that.

Users want to be able to customize 4 color values:
- The header (done)
- The bar (done)
- The purple button color
- The pink button color

At a minimum, those should change in vote, results, and success pages.

**Comment by proggeramlug** [comment]
Okay, I need a bit more explanation here:

Can you send me a screenshot with both colors and for the buttons? I'm only seeing purple right now.

Also, the footer is actually never visible for me, neither when voting nor on the success page. It might be on the results page, but how are we showing the logo then?

**Comment by ggordn3r** [comment]
Yes, sorry, I should have clarified this better and shared a mockup. Pink appears on the back button on mobile and the "View Results" button.

Pink example 1, back button on mobile:
![image](https://user-images.githubusercontent.com/10144772/80853687-a25b4300-8c00-11ea-92ca-22151f8f4e4a.png)

Pink example 2, View results:
![image](https://user-images.githubusercontent.com/10144772/80853696-c454c580-8c00-11ea-8540-17aff2e1bbc3.png)

Here is what the logo should look like at the bottom. To be clear, _the RankIt logo only appears at the bottom when there is a custom logo on top_. I think this footer should be visible on all pages except the success page (because of the purple background), but that doesn't matter for this issue. The logo just needs to be in the footer whenever/wherever the footer appears.
![Logo at bottom mockup](https://user-images.githubusercontent.com/10144772/80853653-61fbc500-8c00-11ea-88ea-3bd6bab2cff6.png)

**Comment by proggeramlug** [comment]
"UPDATE: the footer is on the results page, it's just hiding behind the buttons 😆 Please fix that:"

how do we want to fix this? The problem is that the buttons are statically fixed to the bottom here and they will lay over anything else. So how would we keep the footer visible? @ggordn3r

**Comment by ggordn3r** [comment]
Hmm... dumb solution: add 2-3 line breaks to the bottom of the footer? If we have blank space the size of the buttons underneath then the footer will be visible when someone scrolls all the way down.

**Comment by proggeramlug** [comment]
Check it out now!

**Comment by ggordn3r** [comment]
Still not quite visible on my mobile screen. Add 2 more lines and let's try again.

**Comment by proggeramlug** [comment]
try now

**Comment by ggordn3r** [comment]
That looks perfect. Now I can get back to testing the actual functionality. I'm using this poll to test: https://rankit.skelpo.com/results/D48E57IxOKn4kFnCCgwJ/summary

2 small fixes to the success page:
- Use button color 2 for the CTA button. When not custom set, the default should be the pink button color we use elsewhere
- Notice the sharing buttons are being cut off like the footer was before. Can that be fixed the same way, by adding line breaks underneath?
![image](https://user-images.githubusercontent.com/10144772/81101124-a7a7de80-8edb-11ea-9210-c0f859821ab7.png)

I modified all of the colors (font, button 1, button 2) to tan and orange but the regular colors appear, even if I force the cache to refresh:
![image](https://user-images.githubusercontent.com/10144772/81101558-53e9c500-8edc-11ea-8219-d54baf813b30.png)

I intentionally tested the poll with a gigantic logo image. It's handled fine on desktop but grows huge on mobile. Let's keep the size of the logo the same on mobile as it is on desktop, i.e. regardless of screen width.
![Screenshot_20200505-142727_Chrome](https://user-images.githubusercontent.com/10144772/81102024-ff931500-8edc-11ea-821a-1bc54b100f3b.jpg)

The ability to modify the branding should be available to super users and admins, but not regular users. I've confirmed that the permissions work properly for both regular users and admins, but I realized that I may have missed a requirement here: is there currently a "super user" type between user and admin?

If not, a super user should be able to do everything a user can do, plus customize the branding. That's the only difference for now. They may receive more powers in the next release.
